x-pack/legacy/plugins/apm/public/components/app/ServiceMap/Popover/index.tsx

```diff
@@ -62,65 +77,25 @@ export function Popover({ focusedServiceName }: PopoverProps) {
     };
   }, [cy]);
 
-  const renderedHeight = selectedNode?.renderedHeight() ?? 0;
-  const renderedWidth = selectedNode?.renderedWidth() ?? 0;
-  const { x, y } = selectedNode?.renderedPosition() ?? { x: 0, y: 0 };
-  const isOpen = !!selectedNode;
-  const selectedNodeServiceName: string = selectedNode?.data('id');
-  const isService = selectedNode?.data('type') === 'service';
-  const triggerStyle: CSSProperties = {
-    background: 'transparent',
-    height: renderedHeight,
-    position: 'absolute',
-    width: renderedWidth
-  };
-  const trigger = <div className="trigger" style={triggerStyle} />;
-
-  const zoom = cy?.zoom() ?? 1;
-  const height = selectedNode?.height() ?? 0;
-  const translateY = y - (zoom + 1) * (height / 2);
-  const popoverStyle: CSSProperties = {
-    position: 'absolute',
-    transform: `translate(${x}px, ${translateY}px)`
-  };
-  const data = selectedNode?.data() ?? {};
-  const label = data.label || selectedNodeServiceName;
+  // Handle positioning of popover. This makes it so the popover positions
+  // itself correctly and the arrows are always pointing to where they should.
+  useEffect(() => {
+    if (popoverRef.current) {
+      popoverRef.current.positionPopoverFluid();
+    }
+  }, [popoverRef, x, y]);
 
   return (
-    <EuiPopover
-      anchorPosition={'upCenter'}
+    <Contents
```

Comment:
The naming is a little confusing to me. It seems like the Contents component renders the actual EuiPopover not just the contents of the popover.

Reply:
Yeah ok you're right. I'll leave the `EuiPopover` in this file and put the rest in contents.

I really just want to separate the positioning logic from what goes inside, as well as make it viewable in storybook.